### PR TITLE
Adding newe2e supportability for testing for azcopy list output-type=text

### PR DESCRIPTION
**Pull request by siminsavani-msft**
Added back text output to azcopy list missed in https://github.com/Azure/azure-storage-azcopy/pull/2583 + newe2e testing supportability for text output 

Resolves https://github.com/Azure/azure-storage-azcopy/issues/2709 & https://github.com/Azure/azure-storage-azcopy/issues/2694

### CODE REVIEW

File: e2etest/newe2e_task_validation.go [hunk 1]
```diff
@@ -191,3 +194,110 @@ func checkMultipleErrors(errorMsg []string, line string) bool {
...
+func expectedObjectsToString(expectedObjects map[AzCopyOutputKey]cmd.AzCopyListObject) []string {
+	var stringArray []string
+
+	for _, val := range expectedObjects {
+		stringArray = append(stringArray, toString(val))
+	}
+	return stringArray
+}
+
+func toString(lo cmd.AzCopyListObject) string {
```
Comment (gapra-msft) [inline e2etest/newe2e_task_validation.go:261]: Should we move this logic to the list file to live with the AzCopyListObject?

File: e2etest/zt_newe2e_list_test.go [hunk 2]
```diff
@@ -90,6 +130,51 @@ func (s *ListSuite) Scenario_ListRunningTally(svm *ScenarioVariationManager) {
 	ValidateListOutput(svm, stdout, expectedObjects, expectedSummary)
 }
 
+func (s *ListSuite) Scenario_ListVersionIdNoAdditionalVersionsText(svm *ScenarioVariationManager) {
```
Comment (gapra-msft) [inline e2etest/zt_newe2e_list_test.go:133]: If we are testing the text output for all the properties, we should test more properties than just 1 I think.

**Review by gapra-msft — COMMENTED** [review]
left a few comments.

File: e2etest/newe2e_task_validation.go [hunk 3]
```diff
@@ -191,3 +194,110 @@ func checkMultipleErrors(errorMsg []string, line string) bool {
...
+		a.Error(fmt.Sprintf("Does not match - raw:%s. expectedObj:%s.", rawObj, expectedObj))
+	}
+
+	// if expectedObj summary is provided, the last two elements of raw are part of the summary
+	if expectedSummary != nil {
+		validateSummary(a, rawSum, expectedSummary)
+	}
+}
+
+func cleanupRaw(raw []string, hasSummary bool) ([]string, []string) {
```
Comment (adreed-msft) [inline e2etest/newe2e_task_validation.go:220]: `cleanupRaw` is not a super great function name either. I wouldn't describe what's happening here as "cleaning up", and this function is an extremely verbose (and slow!) way to do `output, summary := raw[:len(raw)-summaryLength], raw[len(raw)-summaryLength]`.

File: e2etest/newe2e_task_validation.go [hunk 4]
```diff
@@ -191,3 +194,110 @@ func checkMultipleErrors(errorMsg []string, line string) bool {
...
+	if rawSummary[0] != expectedSum[0] {
+		a.Error(fmt.Sprintf("File count does not match - raw:%s. expected:%s.", rawSummary[0], expectedSum[0]))
+	}
+
+	if rawSummary[1] != expectedSum[1] {
+		a.Error(fmt.Sprintf("Total file size does not match - raw:%s. expected:%s.", rawSummary[1], expectedSum[1]))
+	}
+}
+
+func expectedObjectsToString(expectedObjects map[AzCopyOutputKey]cmd.AzCopyListObject) []string {
```
Comment (adreed-msft) [inline e2etest/newe2e_task_validation.go:252]: ditto on the stringer comment

File: e2etest/newe2e_task_validation.go [hunk 5]
```diff
@@ -191,3 +194,110 @@ func checkMultipleErrors(errorMsg []string, line string) bool {
 
 	return false
 }
+
+func ValidateListTextOutput(a Asserter, stdout AzCopyStdout, expectedObjects map[AzCopyOutputKey]cmd.AzCopyListObject, expectedSummary *cmd.AzCopyListSummary) {
```
Comment (adreed-msft) [inline e2etest/newe2e_task_validation.go:198]: This works, but could be better.

Consider this-- Parse the output line by line, compare to the map that already exists. If any elements of the map have not been hit, or any of the elements of the output miss the map, log a failure.

File: e2etest/newe2e_task_validation.go (hunk 1 above)
Comment (adreed-msft) [reply e2etest/newe2e_task_validation.go:261]: Nitpick: This is also code duplication from AzCopy (`list.go:352-410`).

File: e2etest/zt_newe2e_list_test.go (hunk 2 above)
Comment (adreed-msft) [reply e2etest/zt_newe2e_list_test.go:133]: Ditto

**Review by adreed-msft — CHANGES_REQUESTED** [review]

**Comment by gengyujie** [comment]
Hello Team, may I know when this issue will get fixed, still in 10.25.0, azcopy list command still doesn't work. it won't list the content.

image

**Comment by siminsavani-msft** [comment]
> Hello Team, may I know when this issue will get fixed, still in 10.25.0, azcopy list command still doesn't work. it won't list the content.
> 
> image

Hi @gengyujie ! Thanks for reaching out, we are doing our best to get this out as soon as we can! Apologies for the delay!

**Review by adreed-msft — CHANGES_REQUESTED** [review]
This is *significantly* better than the first iteration. Nice work!

**Review by adreed-msft — APPROVED** [review]

**Review by adreed-msft — APPROVED** [review]

**Comment by adreed-msft** [comment]
(rq changes was accidental)

**Comment by gengyujie** [comment]
Hello Team, v10.25.1 list command seems don't add INFO in the output. but previous version have this :
image
image

**Pull request merged**